src/structures/Widget.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const Base = require('./Base');
+const Collection = require('../util/Collection');
+
+/**
+ * Represents a Widget.
+ */
+class Widget extends Base {
+  /**
+   * @param {Client} client - The instantiating client
+   * @param {Object} data - The raw data;
+   */
+  constructor(client, data) {
+    super(client);
+    this._patch(data);
+  }
+  /**
+   * Builds the widget with the provided data.
+   * @param {*} data The raw data of the widget
+   * @private
+   */
+  _patch(data) {
+    /**
+     * When a widget is disable, data will always send a code. When it's enable, data doesn't have a code.
+     */
+    if ('code' in data) {
+      /**
+       * The code provide with the data.
+       * @type {?number}
+       */
+      this.code = data.code;
+
+      /**
+       * The error message.
+       * @type {?string}
+       */
+      this.message = data.message;
+
+      /**
+       * Check if the guild has the widget enabled or not.
+       * @type {boolean}
+       */
+      this.enabled = false;
```

Comment:
"enabled" is the only thing who is everytime set, it's to help people for checking if the guild has the widget enabled or not